gigascience/gigadb-website
# Create private dataset page

# Pull request for issue: #1023

This is a pull request for the following functionalities:
1. A new dataset `100039` with dataset type `Incomplete` is introduced.
2. Upload status `Incomplete` has been included in the admin dataset page.
3. Dataset with upload status `AuthorReview` can be created with private URL.
4. `Create/Reset Private URL` is enable for the  page type `hidden` and `draft` datasets.
5. The private URL will contain `/dataset/DOI/token/$token`. 

## How to test?
1. Dataset page `http://gigadb.gigasciencejournal.com:9170/dataset/100039` will not be working because it is not a public page.
2. Go to `http://gigadb.gigasciencejournal.com:9170/adminDataset/update/id/5`,  you should see upload status is `Incomplete`, and `Create/Reset Private URL` button is available at the bottom. When you click `Create/Reset Private URL` button, you will be redirected to, for example, `http://gigadb.gigasciencejournal.com:9170/dataset/100039/token/LN9aWzJ_y7FpAWv7`.
If you go back to `http://gigadb.gigasciencejournal.com:9170/adminDataset/update/id/5`, you should see `Open Private URL` link is available as well.
3. `Create Dataset` in `http://gigadb.gigasciencejournal.com:9170/adminDataset/admin` page, after filling in all mandatory fields, dataset with upload status `AuthorReview` will be created and redirected to private URL, and buttons `Create/Reset Private URL` and `Open Private URL` will then be available for that dataset.

## How have functionalities been implemented?

In `protected/controllers/AdminDatasetController.php`, `hidden` and `draft` page type are explicitly  included  in the `actionPrivate`,  upload status `Incomplete` belongs to hidden page, while `AuthorReview` belongs to draft page, Create/Reset Private URL` button can be triggered and then create a private url with random 16 chars token. Finally, the page will be redirected to `/dataset/DOI/token/$token`.

In `protected/views/adminDataset/_form.php`, make `Create/Reset Private URL` and `Open Private URL` buttons available for creating new `AuthorReview` dataset.

## Any issues with implementation?
None.

## Any changes to automated tests?

A new test dataset is needed, if any current dataset in `data/dev` changes its upload status, the related scenarios need to be updated, this will shrink the code coverage.
So a new dataset `100039` is introduced to `data/dev` with its upload status is `Incomplete`. Therefore, when creating acceptance tests, current passing scenarios will not be affected.


## Any changes to documentation?
None.

## Any technical debt repayment?

A draft PR was created to remove extra template files #1027, now the fix is attached to this PR.

Before the fix, unnecessary excel files will be downloaded to the file systems and they need to be removed manually.

In ` tests/functional/DownloadTemplateFilesCest.php`,  an `after` function is added to check for the existence of the excel file and then remove it after each test, so the state of the file systems could remain as before the test.


## Any improvements to CI/CD pipeline?
None.
